### PR TITLE
Update translation of 1-js/04-object-basics/01-object/article.md

### PR DESCRIPTION
**Pull request by leviding**
#### 所做更改如下：

文件名 | 参考上游 commit | 更改（理由）
-|-|-
article.md | https://github.com/javascript-tutorial/en.javascript.info/commit/b0e67489deee1123dcea2d2b156867d1dbfbc756 | 更新译文

其他修改的文件都是极其不频繁的图片更新，就不一一赘述了。

### CODE REVIEW

File: 1-js/04-object-basics/01-object/article.md [hunk 1]
```diff
@@ -444,9 +461,9 @@ for(let code in codes) {
 
 ## 引用复制
 
-对象和其他原始的类型相比有一个很重要的区别，对象都是按引用存储复制的。
+对象和其他原始类型的一个根本的区别是，对象都是“通过引用”存储和复制的。
 
-原始类型：字符串，数字，布尔类型 -- 是被整个赋值的。
+原始类型：字符串，数字，布尔类型 — 作为整体值被复制或复制。
```
Comment (bemself) [inline 1-js/04-object-basics/01-object/article.md]:
```suggestion
原始类型：字符串，数字，布尔类型 — 作为整体值被赋值或复制。
```

File: 1-js/04-object-basics/01-object/article.md [hunk 2]
```diff
@@ -692,42 +709,42 @@ user.sizes.width++;       // 在这里改变一个属性的值
...
 
-有一个标准的深拷贝算法，解决上面和一些更复杂的情况，叫做 [Structured cloning algorithm](https://html.spec.whatwg.org/multipage/structured-data.html#safe-passing-of-structured-data)。为了不重复造轮子，我们使用它的一个 JS 实现的库 [lodash](https://lodash.com), 方法名叫做 [_.cloneDeep(obj)](https://lodash.com/docs#cloneDeep)。
 
 ## 总结
 
 对象是具有一些特殊特性的关联数组。
 
-他们存储键值对，其中：
-- 属性的键必须是字符串或者Symbol（通常是字符串）。
+他们存储属性（键值对），其中：
```
Comment (bemself) [inline 1-js/04-object-basics/01-object/article.md]:
```suggestion
它们存储属性（键值对），其中：
```

File: 1-js/04-object-basics/01-object/article.md [hunk 3]
```diff
@@ -692,42 +709,42 @@ user.sizes.width++;       // 在这里改变一个属性的值
...
 其他操作：
 - 删除属性：`delete obj.prop`。
-- 检查属性是否存在：`"key" in obj`。
+- 检查是否存在给定键的属性：`"key" in obj`。
 - 遍历对象：`for(let key in obj)` 循环。
 
-对象根据引用来赋值或者复制。换句话说，变量存的不是对象的"值"，而是值的 "引用"（内存地址）。
-所以复制变量或者传递变量到方法中只是复制了对象的引用。
-所有的引用操作（像增加，删除属性）都作用于同一个对象。
+对象是通过引用被赋值或复制的。换句话说，变量存储的不是对象的值“”，而是值的“引用”（内存地址）。所以复制这样的变量或者将其作为函数参数进行传递时，复制的是引用，而不是对象。基于复制的引用（例如添加/删除属性）执行的所有的操作，都是在同一个对象上执行的。
```
Comment (bemself) [inline 1-js/04-object-basics/01-object/article.md]:
```suggestion
对象是通过引用被赋值或复制的。换句话说，变量存储的不是”对象的值“，而是值的“引用”（内存地址）。所以复制这样的变量或者将其作为函数参数进行传递时，复制的是引用，而不是对象。基于复制的引用（例如添加/删除属性）执行的所有的操作，都是在同一个对象上执行的。
```

File: 1-js/04-object-basics/01-object/article.md [hunk 4]
```diff
@@ -692,42 +709,42 @@ user.sizes.width++;       // 在这里改变一个属性的值
...
-- `Date` 存储时间日期。
-- `Error` 存储错误信息
-- ...等等。
+- `Array` 用于存储有序数据集合，
+- `Date` 用于存储时间日期，
+- `Error` 用于存储错误信息。
+- ……等等。
 
-他们有一些特别的特性，我们将在后面学习到。有时候大家说“数组类型”，“时间类型”，他们都属于对象类型的一种，都以不同的方式对对象类型做了一些扩展。
+他们有着各自特别的特性，我们将在后面学习到。有时候大家会说“数组类型”或“时间类型”，但其实他们并不是自己的类型，而是属于对象类型的一种。他们以不同的方式对对象类型做了一些扩展。
```
Comment (bemself) [inline 1-js/04-object-basics/01-object/article.md]:
```suggestion
它们有着各自特别的特性，我们将在后面学习到。有时候大家会说“数组类型”或“日期类型”，但其实它们并不是自身所属的类型，而是属于一个对象类型即 "object"。它们以不同的方式对 "object" 做了一些扩展。
```

Comment (bemself) [reply 1-js/04-object-basics/01-object/article.md]: 原文：这里我的理解是， single "object" data type.是指所有哪些数组、日期类型的对象都是继承自 object 类型。

Sometimes people say something like "Array type" or "Date type", but formally they are not types of their own, but belong to a single "object" data type. And they extend it in various ways.

**Comment by leviding** [comment]
@bemself 修改完了，给力！

**Comment by bemself** [comment]
@leviding 👻 可以帮忙看下为啥状态还是 change requested 吗？我没找到还有哪里...

**Comment by leviding** [comment]
@bemself 你 approve 一下就好了

**Pull request merged**